### ISSUE TITLE
Pagination not working for POST /search if datetimes not the same for all items (pgstac)

### ISSUE THREAD

**Issue by zstatmanweil** [stac-utils/stac-fastapi#249]
When the `limit` is less than the total number of items in a collection in a POST request to `/search` with the pgstac backend, the total number of items returned if you loop through `search.get_items()` or `search.get_all_items()` is incorrect. This is not happening with the `joplin` test data and a relevant test (`test_pagination_item_collection`) because all of the items in both those instances have the same datetime. 

I found when I ran the service locally (up-to-date repo) and copied the joplin data to a new collection but varied the dates, this issue exists. Here is the code I used to do that in case anyone wants to duplicate. Note that while I did use `pystac_client` in my example below, I confirmed it is not in issue with that library - the same thing happens with just requests. 

```python
import requests
from pystac_client import Client
import uuid

host = "http://0.0.0.0:8082"
collection = "joplin"
new_collection = "joplin-diff-dates"

# Create a new collection
resp = requests.get(f"{host}/collections/{collection}")
col = resp.json()
col.pop("links")
col["id"] = new_collection
resp = requests.post(f"{host}/collections", json=col)

# Copy items from existing to new collection with different dates
cat = Client.open(host, ignore_conformance=True)
search = cat.search(collections=[collection])
for idx, i in enumerate(search.get_items()):
    i.id = str(uuid.uuid4())
    i.collection_id = new_collection
    d = i.to_dict()
    d.pop("links")    
    d["properties"].update(datetime=f"2020-01-{idx+1:02d}T00:00:00")
    resp = requests.post(f"{host}/collections/{i.collection_id}/items", json=d)
    resp.raise_for_status()
```

Then when you search the collection, you get different results back (should be 30):
```python
search = cat.search(collections=[new_collection], limit=10)
len(list(search.get_items()))
```

limit of 10 yields 11 results
limit of 15 yields 22 results
limit of 20 yields 29 results
limit of 50 yields 30 results

I am also able to break the `test_pagination_item_collection` collection in the `test_item.py` file if I add this line in the ingest five items portion:
`      test_item["properties"]["datetime"] = f"2020-01-0{idx+1}:T00:00:00Z"`

I realize this may be an issue with pgstac and not stac-fastapi but haven't been able to confirm that yet.

**Comment by zstatmanweil** [comment]
Oh, also confirmed this isn't an issue with sqlalchemy backend.

**Comment by vincentsarago** [comment]
cc @bitner

**Comment by zstatmanweil** [comment]
This was fixed with [pgstac v3.3](https://github.com/stac-utils/pgstac/blob/main/CHANGELOG.md) and stac-fastapi now points to that version. Closing issue!
